src/filter/mod.rs

```diff
@@ -143,6 +143,21 @@ fn pretty2(op: &Op, indent: usize, compose: bool) -> String {
     }
 }
 
+pub fn nesting(filter: Filter) -> usize {
+    nesting2(&to_op(filter))
+}
+
+fn nesting2(op: &Op) -> usize {
+    match op {
+        Op::Compose(filters) => 1 + filters.iter().map(|f| nesting(*f)).fold(0, |a, b| a.max(b)),
+        Op::Exclude(filter) => 1 + nesting(*filter),
+        Op::Workspace(_) => usize::MAX,
```

Comment:
Is this to avoid parsing workspaces?

Reply:
Kind of, yes. We can't parse them because we don't know the content. But it does not really matter because we don't support nested workspaces anyway, so they should always be last.